OpenRA.Game/Map/Map.cs

```diff
@@ -285,6 +286,24 @@ public static string ComputeUID(IReadOnlyPackage package)
 			}
 		}
 
+		public static int GetMapFormat(IReadOnlyPackage p)
+		{
+			Dictionary<string, MiniYaml> yaml;
+			using (var yamlStream = p.GetStream("map.yaml"))
+			{
+				if (yamlStream == null)
+					throw new FileNotFoundException("Required file map.yaml not present in this map");
+
+				yaml = new MiniYaml(null, MiniYaml.FromStream(yamlStream, "map.yaml", stringPool: Game.ModData.MapCache.StringPool)).ToDictionary();
```

Comment:
Parsing the full file to yaml just to extract `MapFormat` has an unnecessarily large overhead.

IMO a better solution would be to use a regex to match and extract the format directly.

Reply:
Sounds like a better solution